### PR TITLE
This removes PEEP control, and wires up the controller with the hardware

### PR DESCRIPTION
**Pull request by mschottdorf**
1. Fixed the hardware issues, i.e. correct pressure measurements.
2. Allows closed loop PID and state control.
3. Default is PID control (seems to work better). On the dinky with these results:

image

Note: Installed was a 5 cm H2O PEEP valve, so this is not software controlled.

This is from a different run, where you can see control-in, and fast out
image

### CODE REVIEW

File: tests/test_controller.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def test_control_dynamical(control_type):
 
     vals_start = Controller.get_sensors()
 
-    v_peep = random.randint(5, 10)
+    v_peep = 5
```
Comment (mschottdorf) [inline tests/test_controller.py:105]: this should change, and incorporate the results of our meeting tomorrow

File: tests/test_timeoutdecorator.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def foo(sleeptime):
         t0 = time.time()
         foo(dt)
         t1 = time.time()
-        assert t1-t0 < dt*1.5
+        assert t1-t0 < dt*2
```
Comment (mschottdorf) [inline tests/test_timeoutdecorator.py:23]: Increased this a bit, as it threw sporadic errors on my laptop

File: vent/controller/control_module.py [hunk 3]
```diff
@@ -397,8 +397,12 @@ def __test_for_alarms(self):
                                   value=self._DATA_PRESSURE)
             if time.time() - self.HAPA.start_time > self.cough_duration:       # 100 ms active to avoid being triggered by coughs
                 self.__SET_PIP = 30                 # Default: PIP to 30
-                self.__control_signal_out = np.inf
-                self.__control_signal_in  = 0
+                for i in range(5):                   # Make sure to send this command for 100ms -> release pressure immediately
```
Comment (mschottdorf) [inline vent/controller/control_module.py:400]: this is new. Sometimes a valve might be stuck, so the first thing is making sure it is open, by sending open command multiple times

File: vent/controller/control_module.py [hunk 4]
```diff
@@ -828,20 +834,20 @@ def _start_mainloop(self):
         # # get final values on stop
         self._controls_from_COPY()  # Update controls from possibly updated values as a chunk
         self._sensor_to_COPY()  # Copy sensor values to COPY
-
+        self._set_HAL(valve_open_in = 0, valve_open_out = 1)  # Defined state to make sure that it does not pop up.
```
Comment (mschottdorf) [inline vent/controller/control_module.py:837]: this is also a new safety line, making sure the valves are all closed

**Comment by zajdel** [comment]
Results look very nice. Looks like you're hitting the PIP fast and reliably!

File: tests/test_controller.py (hunk 1 above)
Comment (zajdel) [reply tests/test_controller.py:105]: When therapists set a PEEP mechanical valve, they probably would like some feedback on the GUI on what the effective PEEP is. Perhaps the user does not set "desired PEEP" as a monitor, but rather sees the "average PEEP" over the last several cycles, so they can adjust slightly if desired.

**Comment by zajdel** [comment]
Also physically tested on the Dinky. I would say this is ready to merge.

**Pull request merged**